community/index/src/main/java/org/neo4j/index/gbptree/ConsistencyChecker.java

```diff
@@ -164,6 +190,47 @@ private void assertKeyOrder( PageCursor cursor, KeyRange<KEY> range )
         }
     }
 
+    static void assertNoCrashOrBrokenPointerInGSPP( PageCursor cursor, long stableGeneration, long unstableGeneration,
+            String pointerFieldName, int offset, TreeNode<?,?> treeNode )
+    {
+        cursor.setOffset( offset );
```

Comment:
The structure as a whole of this class as well as TreePrinter needs to be redesigned to better be able to properly read from page cursor. Although this is a test utility and works for what it does for the moment. Actually, I'll move them to src/test for the time being.